src/coreclr/jit/sideeffects.cpp

```diff
@@ -148,6 +148,16 @@ AliasSet::NodeInfo::NodeInfo(Compiler* compiler, GenTree* node)
         }
 
         m_flags = ALIAS_READS_ADDRESSABLE_LOCATION | ALIAS_WRITES_ADDRESSABLE_LOCATION;
+
+        // For calls having return buffer, update the local number that is written after his call.
```

Comment:
This should be done before the "is pure" check, as purity does not consider the ABI.

Also, it needs to set `ALIAS_WRITES_ADDRESSABLE_LOCATION` in case the local is address-exposed.

Reply:
The local won't be address-exposed in case of "return buffer" so not sure which scenario are you referring?

Reply:
We don't know that -- it can still get exposed if the address escaped to something other than the call that defined it.

Reply:
Realized that Lowering converts the arg into `PutArg` and here, we might see `PutArg`.